### ISSUE TITLE
[V2] is it fair to recommend people to move to v2 while it's mostly broken and not ready for production?

### ISSUE THREAD

**Issue by terreb** [wix/react-native-navigation#3867]
### Issue Description

I moved from v1 to v2 since the docs recommend to do so, plus I had some issues with the v1 that won't be never addressed since the devs had dropped support. But those issues were not that crucial (code push didn't work properly and e2e tests) the navigation itself worked. Now when I switched to v2 I've been encountering dozens of issues and crashes. Almost everything is broken if I try to apply styles or do something a bit more sophisticated than a usual stack of screen. TopBar, BottomBar, Drawer are all not working properly. I see multiple other issues have been created on GitHub and most of them get label stale. I do understand that devs are busy with developing the plugin and can not react on all the issues.

My only question is why you recommend to migrate/start with v2 while it's in alpha and far from being ready to use in production and not mentioning this anywhere in docs?

**Comment by pakyo-pan** [comment]
Many issues, please fix as soon as possible.

**Comment by ItsNoHax** [comment]
I would also like an update on: https://github.com/wix/react-native-notifications

I'm stuck right now because I can't implement push notifications.

**Comment by santanu4ver** [comment]
I wonder if dev even has enough energy/wish to address this question. Also started thinking if going back to v1 would save me some time to do actual development other than wasting on nasty bugs in v2 when most of them not sure when to be address.

**Comment by jengo** [comment]
I've been learning React Native and I have been spinning my wheels just trying to get any basic navigation to work.  I've been back and forth on libraries.  I've found v2 to be horribly broken for me, and so little documentation that it's impossible to make anything look and work correctly.

I know the devs are hard at work on v2 and I am looking forward to a stable version. Personally, I think it's way too premature to suggest using v2 in the documentation.  Not until it's at least beta level quality.

**Comment by ericketts** [comment]
its also disappointing that there's a freeze on accepting PRs for V1, so a PR I put up fixing crashes caused by `ReactInstanceManager.recreateReactContextInBackground()` just sat there until the bot marked it stale and eventually closed it. 


this is really something that needs to be part of react-native core for it to ever be a serious replacement for native development (and you know that facebook almost certainly has at least one in-house nav lib, likely with a high level of polish given the amount of man hours they're able to devote to such a thing...)

**Comment by FeniXenAaaa** [comment]
Well, even if new version has lack of documentation and is unstable it does make sense to start new project with it. Because you don't want to start brand new project with something that is outdated and probably won't be maintained in future because of fundamental problems, right?

**Comment by terreb** [comment]
@FeniXenAaaa, yes makes sense, but only if you plan to have a production release in say 6 months or so, and if you have plenty of spare time trying to workaround/fix dozens of bugs. Sure you may be lucky enough that with your requirements for the navigation it just works for you.

**Comment by FeniXenAaaa** [comment]
@terreb yeah, probably they could be more precise on what version when to use. But you as an experienced developer with RNN V1 should understand that V2 is in alpha and it is absolutely new because it has new fundamental approaches therefore it will require a lot of effort to get familiar with it. And of course it shouldn't be a choice when you have release less than in a few month. It is obvious that if you need to bootstrap something quickly a tool that is in active phase of development (without stable releases so far) shouldn't be considered to use.

**Comment by terreb** [comment]
@FeniXenAaaa, this is great if reading "New to RNN? We recommend you start with v2" all those things you've just written are obvious for you;) I doubt any one else new to RNN reading their statement understands how far the current state from a stable release and don't know that after installing v2 gets involved in an endless game on how to make work one thing without breaking another.

**Comment by rat-moonshine** [comment]
Wix doing a great work to help the community roll the RN stuff, which is appreciable. Along with that (IMO) to roll the transition smoothly, it could have been great if they offered following things (as one already suggested by one at here):

1. Accepting nasty v.1.0 bugs
2. Having some lines added pointing possible breaks if go by v.2.0 as [Airbnb](https://github.com/airbnb/native-navigation) did! So people having a timeframe to release by next 6 months their project, can decide between things:

> React Native Navigation by Wix engineering is an alternative library that uses "Native" navigation components of each platform, and has been around longer than Native Navigation. If you need a stable / production-ready navigation library today that uses native platform based navigation components, we recommend you check this library out.

**Comment by ethaizone** [comment]
I think I found many problems after I use V2. In case XCode 10 I think it's not main issue because this update is not long and any library can break from its change too but I found another issue that shouldn't found on V2.

Example this pull-request #3833 that I tried add onNewIntent on v2. OK. I tried make pull-request but it can't merge because some not stable branch and _No any info that which branch that I should make contribute_.

It's not really important except one thing. _**Why do I need to create that pull-request at beginning?** On V1 we have onNewIntent normally and this is really basic method that all React Native project should have it. Why does it disppear from v2?_

It's question in my head that v2 is really ready for production or not. If I answer this question, **I think it not for production right now.**

We can't blame anyone. It's just that we place trust on wrong place.

**Comment by howg0924** [comment]
I recommend people with tabBar flicker issue to give pribeh's method a try:

https://github.com/wix/react-native-navigation/issues/2184#issuecomment-397830334

It solved the issue for me. The main concept is to give the screen with tabs a fixed height.

@adamterlson

**Comment by stokesbga** [comment]
React native is, at its core, a hack. Trust those of us who have been building with it since 2015, there will never be a solution that just works 100% of the time. RNN V2 is extremely frustrating and stressful to migrate to, but you can get a working solution. After a LOT of refactoring, quick fixes, and separating android and ios components, it will work. It is no more or less stable than V1 imo, and I'll kill myself before I go to react-navigation. That lib is maintained by assholes... and worse, facebook.

**Comment by putuyoga** [comment]
need suggestion, for new project, is it worth it to use directly on the RNN v2?

**Comment by santanu4ver** [comment]
@putuyoga You really asked this after reading all above 'user experiences' (?) 

[I am already a dead man]

**Comment by thinklinux** [comment]
I started a new project with V2. I didn't have a lot of big issues so far. And the thing is that it's going to be ready sooner or later and I don't want to migrate when my app is ready. At least not that soon. So I prefer going with V2 and opening issues. Also making PRs if I have the time for it.

**Comment by lukebrandonfarrell** [comment]
@thinklinux I agree. V2 has made some significant progress in the past few weeks. If we all come together as a community to spot issues and make pull request it will be a great library in no time.
